partner_industry_secondary/migrations/11.0.1.0.0/post-migration.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2018 Eficent Business and IT Consulting Services, S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+def fill_res_partner_industry(cr):
+    cr.execute(
+        """
+        INSERT INTO res_partner_industry (name, active, create_uid,
+            create_date, write_uid, write_date, %s)
+        SELECT name, TRUE as active, create_uid, create_date, write_uid,
+            write_date, id
+        FROM res_partner_sector
+        """ % openupgrade.get_legacy_name('old_sector_id'),
+    )
+    # corrected parent_id here:
+    cr.execute(
+        """
+        UPDATE res_partner_industry rpi
+        SET parent_id = rpi2.id
+        FROM res_partner_sector rps
+        LEFT JOIN res_partner_sector rps2 ON rps.parent_id = rps2.id
```

Comment:
You don't need this second `res_partner_sector` table.

Just use this on the next `LEFT JOIN`: `ON rps.parent_id = rpi2.%s`